kythe/rust/indexer/src/indexer/analyzers.rs

```diff
@@ -397,32 +397,33 @@ impl<'a, 'b> CrateAnalyzer<'a, 'b> {
         let analysis = self.krate.analysis.clone();
 
         for def in &analysis.defs {
-            let krate_id = self.krate_ids.get(&def.id.krate).ok_or_else(||{
-                KytheError::IndexerError(format!(
-                    "Definition \"{}\" referenced crate \"{}\" which was not found in the krate_ids HashMap",
-                    def.qualname, def.id.krate
-                ))}
-            )?;
-            let krate_signature = format!(
-                "{}_{}_{}",
-                krate_id.disambiguator.0, krate_id.disambiguator.1, krate_id.name
-            );
-
-            // Check for "./" at the beginning and remove it
-            let file_vname = self.file_vnames.get(def.span.file_name.to_str().unwrap());
+            if let Some(krate_id) = self.krate_ids.get(&def.id.krate) {
+                let krate_signature = format!(
+                    "{}_{}_{}",
+                    krate_id.disambiguator.0, krate_id.disambiguator.1, krate_id.name
+                );
+
+                // Check for "./" at the beginning and remove it
+                let file_vname = self.file_vnames.get(def.span.file_name.to_str().unwrap());
+
+                // save_analysis sometimes references files that we don't have as file nodes
+                if file_vname.is_none() {
+                    continue;
+                }
 
-            // save_analysis sometimes references files that we have as file nodes
-            if file_vname.is_none() {
-                continue;
+                // Generate node based on definition type
+                let mut def_vname = self.krate_vname.clone();
+                let def_signature = format!("{}_def_{}", krate_signature, def.id.index);
+                def_vname.set_signature(def_signature.clone());
+                def_vname.set_language("rust".to_string());
+                def_vname.clear_path();
+                self.emit_definition_node(&def_vname, def, file_vname.unwrap())?;
+            } else {
+                eprintln!(
```

Comment:
Diagnostic node here as well?

Reply:
Not possible, we have no signature to create an anchor nor diagnostic node with